plugins/com.google.cloud.tools.eclipse.appengine.standard.java8/src-test/com/google/cloud/tools/eclipse/appengine/standard/java8/AppEngineXsltTransformTests.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.standard.java8;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import com.google.cloud.tools.eclipse.util.Xslt;
+import com.google.common.io.CharStreams;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import javax.xml.transform.TransformerException;
+import org.junit.Test;
+
+/**
+ * Test the appengine-web.xml XSLT transforms
+ */
+public class AppEngineXsltTransformTests {
+
+  @Test
+  public void testAddBare() throws IOException, TransformerException {
+    apply("/xslt/addJava8Runtime.xsl",
+        "<?xml version=\"1.0\" encoding=\"UTF-8\"?><appengine-web-app xmlns=\"http://appengine.google.com/ns/1.0\"/>",
```

Comment:
You don't need the XML declarations assuming this is test input data, and taking them out makes this somewhat simpler. 

You do need it in the output string if you're just doing string comparisons. String comparisons on transformed XML are flaky since there are lot of details such as quotes and empty-element tags that can change without modifying the infoset. A JDK upgrade could break these tests. The simplest way to make this robust is to parse the output and look for the details you want such as a runtime element.